### PR TITLE
HRCPP-283 Added exception if recv read 0 byte

### PR DESCRIPTION
**Pull request by rigazilla**
Added the same behaviour as for the posix socket: throw exception if recv read 0 bytes.

### CODE REVIEW

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 644](http://ci.infinispan.org/viewLog.html?buildId=40641&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 9ac35b4d8688dea52cd6b901cef91f6fd0b55dea
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:00

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 507](http://ci.infinispan.org/viewLog.html?buildId=40639&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 9ac35b4d8688dea52cd6b901cef91f6fd0b55dea
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:22

**Comment by rvansa** [comment]
Looks ok :+1:

File: src/hotrod/sys/windows/Socket.cpp [hunk 1]
```diff
@@ -211,6 +211,10 @@ void Socket::setTimeout(int timeout) {
 
 size_t Socket::read(char *p, size_t length) {
     ssize_t n = recv(fd, p, (int) length, 0);
+	if (n == 0)
```
Comment (slaskawi) [inline src/hotrod/sys/windows/Socket.cpp]: The formatting seems to be inconsistent.

Comment (slaskawi) [inline src/hotrod/sys/windows/Socket.cpp]: Could we add a test for this?

**Comment by rigazilla** [comment]
Fixed indentation and added some curl brakets

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 649](http://ci.infinispan.org/viewLog.html?buildId=41012&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 29de879f1623e8b38ee548b133d49a4bb6b2590e
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:08:54

**Comment by rigazilla** [comment]
Added a test

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL [Build 650](http://ci.infinispan.org/viewLog.html?buildId=41017&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 797f086d7eb1727c9284b02b2738ed05e4aabcd9
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:38

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 514](http://ci.infinispan.org/viewLog.html?buildId=41011&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 797f086d7eb1727c9284b02b2738ed05e4aabcd9
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:18:53

Process exited with code 255

**Comment by tristantarrant** [comment]
Rerunning failed build on Windows

**Comment by tristantarrant** [comment]
Pushed, thanks

**Pull request closed without merging**